src/client/tile.cpp

```diff
@@ -1735,6 +1735,41 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 			baseimg->drop();
 			baseimg = image;
 		}
+		/*
+			[opacity:R
+			Makes the base image transparent according to the given ratio.
+			R must be between 0 and 255.
+			0 means totally transparent.
+			255 means totally opaque.
+		*/
+		else if (str_starts_with(part_of_name, "[opacity:"))
+		{
+			if (baseimg == NULL) {
+				errorstream<<"generateImagePart(): baseimg == NULL "
+						<<"for part_of_name=\""<<part_of_name
+						<<"\", cancelling."<<std::endl;
+				return false;
+			}
+
+			Strfnd sf(part_of_name);
+			sf.next(":");
+			std::string ratio_str = sf.next("");
+
+			int ratio = 0;
+
+			if (is_number(ratio_str))
+				ratio = mystoi(ratio_str, 0, 255);
+
+			core::dimension2d<u32> dim = baseimg->getDimension();
+
+			for (u32 y=0; y<dim.Height; y++)
+			for (u32 x=0; x<dim.Width; x++)
+			{
+				video::SColor c = baseimg->getPixel(x,y);
+				c.setAlpha(ratio);
```

Comment:
What about partwise transparent textures, like the apple? Wouldn't it get a black border because the transparent pixels are set to half-opaque?